scripts/dts-localize.ts

```diff
@@ -0,0 +1,261 @@
+    );
+  }
+
+  // input file could be a relative path to the current path
+  // and desired config could be outside of current cwd folder
+  // so we have to provide absolute path to find config until the root
+  const searchPath = getAbsolutePath(inputFiles[0]);
+  const configFileName = ts.findConfigFile(searchPath, ts.sys.fileExists);
+  if (!configFileName) {
+    throw new Error(`Cannot find config file for file ${searchPath}`);
+  }
+
+  return configFileName;
+}
+
+interface Options {
+  externals: string[];
+  cwd?: string;
+}
+
+export const run = async (entrySourceFiles: string[], outputPath: string, options: Options) => {
+  const compilerOptions = getCompilerOptions(entrySourceFiles);
+  const host = ts.createCompilerHost(compilerOptions);
+  const cwd = options.cwd || process.cwd();
+  const pkg = sync({ cwd }).packageJson;
+  const externals = Object.keys({ ...pkg.dependencies, ...pkg.peerDependencies });
+
+  // this to make paths for local packages as they are in node_modules because of yarn
+  // but it depends on the way you handle "flatting of files"
+  // so basically you can remove this host completely if you handle it in different way
+  host.realpath = (p: string) => p;
+
+  const program = ts.createProgram(entrySourceFiles, compilerOptions, host);
+  const printer = ts.createPrinter({ newLine: ts.NewLineKind.LineFeed, removeComments: false });
+
+  const typeChecker = program.getTypeChecker();
+  const sourceFiles = program.getSourceFiles();
+
+  const filesRemapping = new Map<string, string>();
+  const replaceRemapping = new Map<string, string[]>();
+
+  for (const sourceFile of sourceFiles) {
+    // skip lib files from the compiler
+    if (program.isSourceFileDefaultLibrary(sourceFile)) {
+      continue;
+    }
+
+    // here you can implement any logic regarding how your new files should be placed
+    // as for this POC I used rough solution to use the same structure but I believe it should be improved
+    filesRemapping.set(sourceFile.fileName, getNewPath(cwd, sourceFile.fileName));
+  }
+
+  entrySourceFiles.forEach((file) => {
+    const sourceFile = sourceFiles.find((f) => f.fileName === file);
+
+    actOnSourceFile(sourceFile);
+  });
+
+  function getNewPath(basePath: string, filePath: string) {
+    const relative = path.relative(basePath, filePath);
+    let newPath = '';
+
+    if (relative.includes('node_modules/')) {
+      const [, ...parts] = relative.split('node_modules/');
+      const filename = parts.join('node_modules/').split('/').join('-');
+      newPath = path.join(outputPath, '_modules', filename);
+    } else if (relative.includes('dist/ts-tmp/')) {
+      const [, ...parts] = relative.split('dist/ts-tmp/');
+      const filename = parts.join('').split('/').join('-');
+      newPath = path.join(outputPath, filename);
+    } else {
+      const filename = relative.split('/').join('-');
+      newPath = path.join(outputPath, filename);
+    }
```

Comment:
I think you were going to add a comment explaining this

Reply:
I did!